### PR TITLE
Update JRC hydro database to latest and rerun powerplantmatching

### PR DESCRIPTION
**Pull request by martinhjel**
Closes # 122.

## Change proposed in this Pull Request

Have updated the JRC database to point to the latest version, not version 10 which was built in 2021. 

Have also run the `update_powerplants.py` script to update the `powerplants.csv` file.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other

## Checklist

- [x] I have added a note to release notes `doc/release_notes.rst`.
- [x] I have used `pre-commit run --all` to lint/format/check my contribution
- [x] I have documented the effects of my code changes in the documentation `doc/`.
- [x] I have adjusted the docstrings in the code appropriately.

### CODE REVIEW

**Comment by FabianHofmann** [comment]
Great, thanks @martinhjel! Did you run the full matching process when creating the powerplants.csv?

**Comment by martinhjel** [comment]
Yes, I ran the update_powerplants.py script. It was quite large changes (11,167 additions, 5,403 deletions) maybe you could verify that you get the same changes?

File: powerplantmatching/package_data/config.yaml [hunk 1]
```diff
@@ -89,8 +89,8 @@ ENTSOE-EIC:
   fn: entsoe_eic_codes.csv
 JRC:
   reliability_score: 4
-  fn: jrc-hydro-power-database-10.zip
-  url: https://github.com/energy-modelling-toolkit/hydro-power-database/archive/refs/tags/v10.zip
+  fn: jrc-hydro-power-plant-database.csv
+  url: https://raw.githubusercontent.com/energy-modelling-toolkit/hydro-power-database/master/data/jrc-hydro-power-plant-database.csv
```
Comment (fneum) [inline powerplantmatching/package_data/config.yaml:93]:
```suggestion
  url: https://raw.githubusercontent.com/energy-modelling-toolkit/hydro-power-database/7c25aef/data/jrc-hydro-power-plant-database.csv
```
Let's tie this to a particular commit. The master may change in format any time, and in absence of a tagged release, this is the next best thing.

**Review by fneum — APPROVED** [review]
LGTM. Just to confirm: format has not changed?

**Comment by FabianHofmann** [comment]
Hey @martinhjel, would it be possible for you to reset the powerplants.csv? 
```
git checkout master `powerplants.csv`
```

Then I would update the powerplant separately.

**Comment by fneum** [comment]
- fixed to particlar commit of JRC hydro database
- updated powerplants.csv separately

**Pull request merged**